aten/src/ATen/mps/MPSStream.mm

```diff
@@ -189,21 +190,36 @@ @interface MPSGraphExecutionDescriptor ()
         profiler.beginProfileGPUInterval(mpsGraph);
       }
 
-      if (disableTypeInference) {
-        MPSGraphCompilationDescriptor *compilationDescriptor = [[MPSGraphCompilationDescriptor new] autorelease];
-        [compilationDescriptor disableTypeInference];
-        _executionDescriptor.compilationDescriptor = compilationDescriptor;
+      if (executable) {
+        NSMutableArray *inputsArray  = [NSMutableArray arrayWithCapacity:[feeds count]];
+        NSMutableArray *resultsArray = [NSMutableArray arrayWithCapacity:[results count]];
+        NSUInteger inputIndex = 0, ouputIndex = 0;
+        for (MPSGraphTensor *tensor in [executable feedTensors]) {
+          inputsArray[inputIndex++] = feeds[tensor];
+        }
+        
+        for (MPSGraphTensor *tensor in [executable targetTensors]) {
+          resultsArray[ouputIndex++] = results[tensor];
+        }
+
+        MPSGraphExecutableExecutionDescriptor *executionDescriptor =
+                        [[MPSGraphExecutableExecutionDescriptor new] autorelease];
+
+        executionDescriptor.completionHandler = ^(NSArray<MPSGraphTensorData *> * _Nonnull,
```

Comment:
This is not required (causes unnecessary overhead of calling an empty callback). 
If `executionDescriptor.completionHandler` is nil, the MPSGraph will not run it.

Reply:
Thanks. Updated